File: examples/plot_linear.jl
# plot linear data
# usage: julia plot_linear.jl <config>
import YAML
using Plots
using Mbo.TimeGrid
using Rsvg
plotlyjs()

function main(args)
cfgf = args[1]
dir = dirname(cfgf)
cfg = open(YAML.load, cfgf)
rootname = cfg["rootname"]

# linear plotsa
dat_t = readdlm(joinpath(dir, "$(rootname)_rlin.txt"))
t = dat_t[:,1]
rlin = dat_t[:,2]+1im*dat_t[:,3]
p = plot(t, [real(rlin), imag(rlin)])
png(p, "$(rootname)_rlin.png")

dat_f = readdlm(joinpath(dir, "$(rootname)_slin.txt"))
f = dat_f[:,1]
slin = dat_f[:,2]+1im*dat_f[:,3]
p = plot(f, [real(slin), imag(slin)])
png(p, "$(rootname)_slin.png")
end # function main

main(ARGS)

File: examples/plot_linear.py
#!python
# plot_linear.py: plot linear response in the time and frequency domains.
# usage: python plot_linear.py <config>

import sys
import os.path as pth
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patheffects as pe
import yaml

cfgf = sys.argv[1]

with open(cfgf, "r") as f:
    cfg = yaml.load(f)

fld = pth.dirname(cfgf)
rootname = cfg["rootname"]

t, rr, ri = np.loadtxt(pth.join(fld, "{}_rlin.txt".format(rootname)), unpack=True)
plt.figure()
plt.plot(t, rr)
plt.plot(t, ri)
plt.tight_layout()
plt.savefig("{}_rlin.png".format(rootname), dpi=300)
plt.close()

f, sr, si = np.loadtxt(pth.join(fld, "{}_slin.txt".format(rootname)), unpack=True)
plt.figure()
plt.plot(f, sr)
plt.plot(f, si)
plt.tight_layout()
plt.savefig("{}_slin.png".format(rootname), dpi=300)
